### PR TITLE
doc: add output processing how-to

### PR DESCRIPTION
**Pull request by setrofim**
Add a developer how-to illustrating the use of WA output processing API
by creating an ASCII reporter.

### CODE REVIEW

File: doc/source/developer_information/how_tos/processing_output.rst [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+		  frequency: min
+	    - runtime_params:
+		frequency: max
+	      classifiers:
+		  frequency: max
+	workloads:
+	    - sysbench
+	    - deepbench
+
+This  runs two workloads under two different configurations each -- once with
```
Comment (marcbonnici) [inline doc/source/developer_information/how_tos/processing_output.rst]: Double space after "This"

File: doc/source/developer_information/how_tos/processing_output.rst [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+
+	from collections import defaultdict
+
+	print()
+	print('    Metrics Comparison     ')
+	print('    ------------------     ')
+	print()
+	scores = defaultdict(lambda: defaultdict(lambda: defaultdict()))
+	for job in ro.jobs:
+	for metric in job.metrics:
```
Comment (marcbonnici) [inline doc/source/developer_information/how_tos/processing_output.rst]: Missing indent

File: doc/source/developer_information/how_tos/processing_output.rst [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+		workload = job.label
+		name = metric.name
+		freq = job.classifiers['frequency']
+		for cname, cval in sorted(metric.classifiers.items()):
+		if cname not in job.classifiers:
```
Comment (marcbonnici) [inline doc/source/developer_information/how_tos/processing_output.rst]: Missing indent

File: doc/source/developer_information/how_tos/processing_output.rst [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+.. code-block:: python
+
+	from wa.utils.misc import write_table
+
+	print('           Jobs            ')
+	print('           ----            ')
+	print()
+	rows = []
+	for job in ro.jobs:
+	rows.append([job.id, job.label, job.iteration, job.status])
```
Comment (marcbonnici) [inline doc/source/developer_information/how_tos/processing_output.rst]: Missing indent

File: doc/source/developer_information/how_tos/processing_output.rst [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+		if cname not in job.classifiers:
+			# was not propagated from the job, therefore was
+			# added by the workload
+			name += '/{}={}'.format(cname, cval)
+
+		scores[workload][name][freq] = metric
+
+	rows = []
+	for workload in sorted(scores.keys()):
+	wldata = scores[workload]
```
Comment (marcbonnici) [inline doc/source/developer_information/how_tos/processing_output.rst]: Missing indent

File: doc/source/developer_information/how_tos/processing_output.rst [hunk 6]
```diff
@@ -0,0 +1,394 @@
...
+	for name in sorted(wldata.keys()):
+		min_score = wldata[name]['min'].value
+		max_score = wldata[name]['max'].value
+		delta =  max_score - min_score
+		units = wldata[name]['min'].units or ''
+		lib = wldata[name]['min'].lower_is_better
+
+		warn = ''
+		if (lib and delta > 0) or (not lib and delta < 0):
+		warn = '!!!'
```
Comment (marcbonnici) [inline doc/source/developer_information/how_tos/processing_output.rst]: Missing indent

**Comment by setrofim** [comment]
Comments addressed.

**Pull request merged**